File: src/com/pratham/CircularQueue.java
package com.pratham;

public class CircularQueue {
    static int front = -1;
    static int rear = -1;
    static int size = 5;
    static int[] cq = new int[size];

    public static void main(String[] args) {
        // insertion
        System.out.println("Displaying after insertion");
        enqueue(1);
        enqueue(2);
        enqueue(3);
        enqueue(4);
        enqueue(5);
        display();

        // deletion
        dequeue();
        dequeue();

        System.out.println("Displaying after deletion: ");
        display();

        // again inserting element to check if queue is behaving as circular queue
        enqueue(6);
        enqueue(7);

        // displaying after inserting again
        System.out.println("Displaying after inserting again");
        for (int x : cq) {
            System.out.print(x + " ");
        }
    }

    static void display() {
        for (int i = front; i <= rear; i++) {
            System.out.println(cq[i]);
        }
    }

    static boolean isFull() {
        return front == (rear + 1) % size;
    }

    static boolean isEmpty() {
        return front == -1;
    }

    static void enqueue(int val) {
        if (!isFull()) {
            if (front == -1 && rear == -1) {
                front = 0;
                rear = 0;
            } else if (rear == size - 1 && front != 0) {
                rear = 0;
            } else {
                rear = (rear + 1) % size;
            }
            cq[rear] = val;
        }
    }

    static void dequeue() {
        if (!isEmpty()) {
            int temp = cq[front];
            if (front == rear) {
                front = rear = -1;
            } else if (front == size - 1) {
                front = 0;
            } else {
                front = (front + 1) % size;
            }
        }
    }
}

File: C++ Implementation/circularQueue.cpp
#include<iostream>
using namespace std;
class squeue{
    int front;
    int rear;
    int size;
    int *arr;
    public:
        squeue(int size){
            this->size = size;
            front=rear=-1;
            arr = new int[size];
        }

        bool isFull(){
            
        }

        bool isEmpty(){
           
        }

        void enqueue(int val){
           
        }

        void dequeue(){

        }

        void display(){
            
        }
};
int main()
{
    squeue sq(5);
    sq.enqueue(10);
    sq.enqueue(20);
    sq.enqueue(30);
    sq.enqueue(40);
    sq.enqueue(50);
    sq.display();

    sq.dequeue();
    sq.dequeue();
    sq.display();

    return 0;
}